### PR TITLE
Add .gitignore to the generated projects

### LINKED ISSUE #66: Project generator should include a .gitignore
There's no `.gitignore` in generated projects, which will lead to lots of `__pycache__` and `.pyc` files getting committed to project repos.

Suggest generating one at https://gitignore.io, perhaps like [this](https://gitignore.io/api/osx,linux,python,windows,pycharm) one.

### PR DESCRIPTION
**Pull request by AkshatBajaj**
Fixes #66

### CODE REVIEW

File: templates/new/.gitignore.txt [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Compiled python modules.
+*.pyc
+*$py.class
+__pycache__/
+
```
Comment (ScottyB) [inline templates/new/.gitignore.txt]: Please add `*.ipynb_checkpoints`.

File: templates/new/.gitignore.txt [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+# Setuptools distribution folder.
+/dist/
+
+# Distribution / packaging
+build/
+
```
Comment (ScottyB) [inline templates/new/.gitignore.txt]: We need `./data` as well.

File: templates/new/.gitignore.txt (hunk 1 above)
Comment (AkshatBajaj) [reply templates/new/.gitignore.txt]: Fixed!

File: templates/new/.gitignore.txt (hunk 2 above)
Comment (AkshatBajaj) [reply templates/new/.gitignore.txt]: Good find!

**Pull request merged**